genesyscloud/outbound_campaignrule/resource_genesyscloud_outbound_campaignrule_unit_test.go

```diff
@@ -0,0 +1,376 @@
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleConditions)[0].Parameters.Value, *(*campaignRule.CampaignRuleConditions)[0].Parameters.Value, "campaignRule.CampaignRuleConditions[0].Parameters.Value check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleConditions)[0].Parameters.DialingMode, *(*campaignRule.CampaignRuleConditions)[0].Parameters.DialingMode, "campaignRule.CampaignRuleConditions[0].Parameters.DialingMode check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleActions)[0].Id, *(*campaignRule.CampaignRuleActions)[0].Id, "campaignRule.CampaignRuleActions[0].Id check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleActions)[0].ActionType, *(*campaignRule.CampaignRuleActions)[0].ActionType, "campaignRule.CampaignRuleActions[0].ActionType check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleActions)[0].Parameters.Operator, *(*campaignRule.CampaignRuleActions)[0].Parameters.Operator, "campaignRule.CampaignRuleActions[0].Parameters.Operator check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleActions)[0].Parameters.Priority, *(*campaignRule.CampaignRuleActions)[0].Parameters.Priority, "campaignRule.CampaignRuleActions[0].Parameters.Priority check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleActions)[0].Parameters.Value, *(*campaignRule.CampaignRuleActions)[0].Parameters.Value, "campaignRule.CampaignRuleActions[0].Parameters.Value check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleActions)[0].Parameters.DialingMode, *(*campaignRule.CampaignRuleActions)[0].Parameters.DialingMode, "campaignRule.CampaignRuleActions[0].Parameters.DialingMode check failed in update updateOutboundCampaignruleAttr")
+		assert.Equal(t, *(*testCampaignRule.CampaignRuleActions)[0].CampaignRuleActionEntities.UseTriggeringEntity, *(*campaignRule.CampaignRuleActions)[0].CampaignRuleActionEntities.UseTriggeringEntity, "campaignRule.CampaignRuleActions[0].CampaignRuleActionEntities.UseTriggeringEntity check failed in update updateOutboundCampaignruleAttr")
+		assert.EqualValues(t, *(*testCampaignRule.CampaignRuleActions)[0].CampaignRuleActionEntities.Sequences, *(*campaignRule.CampaignRuleActions)[0].CampaignRuleActionEntities.Sequences, "campaignRule.CampaignRuleActions[0].CampaignRuleActionEntities.Sequences check failed in update updateOutboundCampaignruleAttr")
+		assert.EqualValues(t, *(*testCampaignRule.CampaignRuleActions)[0].CampaignRuleActionEntities.Campaigns, *(*campaignRule.CampaignRuleActions)[0].CampaignRuleActionEntities.Campaigns, "campaignRule.CampaignRuleActions[0].CampaignRuleActionEntities.Campaigns check failed in update updateOutboundCampaignruleAttr")
+
```

Comment:
There must be a cleaner way to accomplish this. This is tough to read and maintain. Maybe translating both objects to json and comparing to see they're equal? Pretty sure there's some package that lets you do that

Reply:
I found this package https://github.com/google/go-cmp that can check to see if the properties of two objects are the same. it will output an empty string if they are, if not it outputs an string showing the differences between the two. I tried it with two callable timesets and this was the output:
```
  platformclientv2.Callabletimeset{
        SetFieldNames: nil,
        Id:            nil,
-       Name:          &"Name 1",
+       Name:          &"Name 2",
        DateCreated:   nil,
        DateModified:  nil,
        Version:       nil,
        CallableTimes: &[]platformclientv2.Callabletime{
                {
                        SetFieldNames: nil,
                        TimeSlots: &[]platformclientv2.Campaigntimeslot{
                                {
                                        SetFieldNames: nil,
-                                       StartTime:     &"12.00.00",
+                                       StartTime:     &"9.00.00",
-                                       StopTime:      &"18.00.00",
+                                       StopTime:      &"10.00.00",
                                        Day:           nil,
                                },
                        },
-                       TimeZoneId: &"Ireland",
+                       TimeZoneId: &"England",
                },
        },
        SelfUri: nil,
  }

```

Reply:
Yeah that looks good 👍 that output string when they're not equal is an ideal error message as well